tests/jupyterhub_api/SessionUtils.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+import requests
+import json
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+import yaml
+import sys
+sys.path.append("..")
+from test_main import Test
+import time
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+class JupyterhubTest(Test):
+    """
+
+    Implements everything required to call Jupyterhub API
+    """
+
+    def __init__(self, port, token, base_path, verify, **kwargs):
+        Test.__init__(self, **kwargs)
+        self.exit = 0
```

Comment:
I wanted to use it as an instance variable, so that I don't have to reinitialise self.exit to zero again (if the function is returning correct results). And if a function is returning wrong results I could simply do self.exit = 1

Reply:
I'm not fully understanding what you're saying... If you want to go to the object and re-check the exit code, it might make sense if you store in the object to avoid having to compute it again. But this should not be defined here, it should instead be defined after call the function that sets the value. And I would do something like:

1. You try to get mytest.exit_code
1. Do you have the value computed?
    * Yes, then return it.
    * No, compute it, store it, then return it.